## 1. Objetivo

Evaluar el riesgo crediticio en una entidad bancaria a través de un modelo predictor.


## 2. Importar librerias

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
import warnings
warnings.filterwarnings('ignore')

## 3. Import train dataset

In [3]:
url = 'https://raw.githubusercontent.com/Ciarzi/DP3_GP1/main/datasets/own_data/df_train.csv'
df = pd.read_csv(url)

## 4. Exploratory Data Analysis

In [4]:
df.shape

(4359, 12)

#ACTUALIZAR
We can see that there are 4359 instances and 12 attributes in the dataset.

### Preview dataset

In [5]:
df.head()

loannumber  longitude_gps  latitude_gps  historial  referido  flag  age  \
0          12       3.432010      6.433055          0         1     1   50   
1           7      11.139350     10.292041          1         1     1   37   
2           3       3.985770      7.491708          0         1     1   44   
3           9       7.457913      9.076574          1         1     1   35   
4           8       3.311981      6.633271          1         1     1   49   

   parte_mes_pago  due_per_day  account_type  education  employment  
0               3         1150             0          3           2  
1               3         1483             0          0           2  
2               2          766             2          0           2  
3               1         1466             0          1           2  
4               2          650             2          0           2

### View summary of dataframe

In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4359 entries, 0 to 4358
Data columns (total 12 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   loannumber      4359 non-null   int64  
 1   longitude_gps   4359 non-null   float64
 2   latitude_gps    4359 non-null   float64
 3   historial       4359 non-null   int64  
 4   referido        4359 non-null   int64  
 5   flag            4359 non-null   int64  
 6   age             4359 non-null   int64  
 7   parte_mes_pago  4359 non-null   int64  
 8   due_per_day     4359 non-null   int64  
 9   account_type    4359 non-null   int64  
 10  education       4359 non-null   int64  
 11  employment      4359 non-null   int64  
dtypes: float64(2), int64(10)
memory usage: 408.8 KB


### View summary statistics of dataframe

In [7]:
df.describe()

loannumber  longitude_gps  latitude_gps    historial     referido  \
count  4359.000000    4359.000000   4359.000000  4359.000000  4359.000000   
mean      5.177564       4.541863      7.254475     0.745125     1.192934   
std       3.655061       7.430102      2.991204     0.435841     0.652024   
min       2.000000    -149.033700    -33.868818     0.000000     0.000000   
25%       2.000000       3.355051      6.474508     0.000000     1.000000   
50%       4.000000       3.584222      6.622782     1.000000     1.000000   
75%       7.000000       6.548874      7.429396     1.000000     1.000000   
max      27.000000     151.209290     71.228069     1.000000     3.000000   

              flag          age  parte_mes_pago  due_per_day  account_type  \
count  4359.000000  4359.000000     4359.000000  4359.000000   4359.000000   
mean      0.782290    37.284240        1.929571   743.598532      1.578802   
std       0.412737     6.169639        0.846039   329.734783      0.806508   
min       0.000000    26.000000        1.000000   366.000000      0.000000   
25%       1.000000    33.000000        1.000000   433.000000      2.000000   
50%       1.000000    37.000000        2.000000   766.000000      2.000000   
75%       1.000000    41.000000        3.000000   816.000000      2.000000   
max       1.000000    60.000000        3.000000  2090.000000      2.000000   

         education   employment  
count  4359.000000  4359.000000  
mean      0.269098     1.650608  
std       0.699988     0.726928  
min       0.000000     0.000000  
25%       0.000000     2.000000  
50%       0.000000     2.000000  
75%       0.000000     2.000000  
max       3.000000     2.000000

### Check for missing values

In [8]:
df.isnull().sum()

loannumber        0
longitude_gps     0
latitude_gps      0
historial         0
referido          0
flag              0
age               0
parte_mes_pago    0
due_per_day       0
account_type      0
education         0
employment        0
dtype: int64

There are no missing values in the dataset.